### ISSUE TITLE
[R] Error: object 'is_eval_err' not found

### ISSUE THREAD

**Issue by UpsideDownRide** [Microsoft/LightGBM#947]
## Environment info
Operating System: Windows  Version	10.0.16294 Build 16294
CPU:  i7-5820K 
C++/Python/R version: R 3.4.2

I have compiled OpenCl version for my system and Windows executable runs just fine. However I'm getting an error in R.

## Error Message:

[LightGBM] [Info] Number of positive: 3140, number of negative: 3373
[LightGBM] [Info] Total Bins 128
[LightGBM] [Info] Number of data: 6513, number of used features: 107
Error: object 'is_eval_err' not found

traceback()

4: merge.eval.string(env)
3: f(env)
2: lgb.train(params, dtrain, nrounds, valids, verbose = verbose, 
       eval_freq = eval_freq, early_stopping_rounds = early_stopping_rounds, 
       init_model = init_model, callbacks = callbacks, ...)
1: lightgbm(data = train$data, label = train$label, num_leaves = 4, 
       learning_rate = 1, nrounds = 2, objective = "binary")

**Comment by huanzhang12** [comment]
This seems to be caused by commit f90dfaf. @bfgray3 Can you take a look at this? I think there is a typo here: https://github.com/Microsoft/LightGBM/blob/master/R-package/R/callback.R#L135, `is_eval_error` should be `is_eval_err`?

@UpsideDownRide Can you change line 135 of `R-package/R/callback.R` into the following:

```
is_eval_err <- length(env$eval_err_list) > 0
```
and see if it works?

**Comment by bfgray3** [comment]
Yes @huanzhang12 you are correct.  Sorry for the inconvenience.

**Comment by bfgray3** [comment]
@huanzhang12 I'm about to put in a PR fixing #946 and #947.

**Comment by huanzhang12** [comment]
@bfgray3 Thanks!

**Comment by huanzhang12** [comment]
Fixed in ac2a914
